overlord/snapstate/backend/snapdata.go

```diff
@@ -68,20 +68,50 @@ func (b Backend) RemoveSnapSaveData(snapInfo *snap.Info, dev snap.Device) error
 	return os.RemoveAll(saveDir)
 }
 
-// RemoveSnapDataDir removes base snap data directory
-func (b Backend) RemoveSnapDataDir(info *snap.Info, hasOtherInstances bool) error {
+// RemoveSnapDataDir removes base snap data directories
+func (b Backend) RemoveSnapDataDir(info *snap.Info, hasOtherInstances bool, opts *dirs.SnapDirOptions) error {
 	if info.InstanceKey != "" {
 		// data directories of snaps with instance key are never used by
 		// other instances
-		if err := os.Remove(snap.BaseDataDir(info.InstanceName())); err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("failed to remove snap %q base directory: %v", info.InstanceName(), err)
+		dirs, err := snapBaseDataDirs(info, opts)
+		if err != nil {
+			return err
+		}
+		var firstRemoveErr error
```

Comment:
This would be the `lastRemoveErr`

Reply:
my preference though would be indeed to record the first error, not the last as that's more consistent with what we do usually